### PR TITLE
Capacity flag per cluster to rate limit tenant resources for che and jenkins

### PR DESCRIPTION
**Pull request by alexeykazakov**
Part of https://github.com/openshiftio/openshift.io/issues/3320

To avoid overloading clusters we need a way to limit total amount of che and jenkins instances running on each cluster. This PR will add a new parameter to the cluster config which will reflect the current capacity of each cluster. When the corresponding secret is modified it will cause the config file update. The file watcher will re-load cluster configuration.

- [x] New capacity parameter `capacity-exhausted` in the cluster configuration and cluster API
- [x] File watcher to update the cluster configuration when /etc/fabric8/oso-clusters.conf is updated
- [x] Tests

For some reason there is no any file update notifications in fabric8.io jenkins environment, so, the tests can't pass. I tested it locally on bare metal, docker, and minishift. It seems to work fine. Tests also pass in centos CI (they run in a docker container). So, I disabled tests in Jenkins file used by fabrict8.io build because these tests don't give us any value anyway. We relay on centos ci builds for tests only.

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @alexeykazakov for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-services/fabric8-auth/69fee3db62f5b99baff16ced79826d7ee90b879f/test-keeper.yml).

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-477-1`

**Comment by surajssd** [comment]
what will this flag be called as? `clusterFull` ? I think it is sensible to do that!

asking because it will be consumed by tenant https://github.com/fabric8-services/fabric8-tenant/pull/583/files#diff-c0b4d0964c7a7f269081aa6229c3ed30R24

**Comment by alexeykazakov** [comment]
@surajssd, @aslakknutsen proposed capacity_exhausted which I like.

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-477-2`

**Comment by alexeykazakov** [comment]
[test]

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-477-5`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-477-6`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-477-7`

**Comment by alexeykazakov** [comment]
[test]

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-477-8`

File: Jenkinsfile [hunk 1]
```diff
@@ -14,15 +14,15 @@ goTemplate{
           dockerOrganisation = 'fabric8'
           project = 'fabric8-auth'
           dockerBuildOptions = '--file Dockerfile.deploy'
-          makeTarget = 'build check-go-format test-unit-junit'
```
Comment (sbose78) [inline Jenkinsfile]: Fair enough, we don't need fabric8 ci infra to test. Only the images are helpful.

**Pull request merged**